extensions/wikia/Email/EmailController.class.php

```diff
@@ -396,6 +406,92 @@ protected function getSocialMessages() {
 		];
 	}
 
+	private function generateMobileApplicationsBadges() {
+		$mobileApplicationsLinks = $this->generateMobileApplicationsLinks();
+		$hasMobileApplicationBadges = count( $mobileApplicationsLinks ) > 0;
+		$badges = [
+			'hasMobileApplicationBadges' => $hasMobileApplicationBadges
+		];
+
+		if ( $hasMobileApplicationBadges ) {
+			if ( $mobileApplicationsLinks['android'] ) {
+				$badges['android'] = [
+					'link' => $mobileApplicationsLinks['android'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-android-badge' )->text() )
+				];
+			}
+			if ( $mobileApplicationsLinks['ios'] ) {
+				$badges['ios'] = [
+					'link' => $mobileApplicationsLinks['ios'],
+					'src' => ImageHelper::getFileUrl(
+						$this->getMessage( 'emailext-mobile-application-ios-badge' )->text() )
+				];
+			}
+		}
+
+		return $badges;
+	}
+
+	private function generateMobileApplicationsLinks() {
+		$mobileApplicationsLinks = [];
+
+		$response = $this->fetchMobileApplicationsDetails();
+
+		if ( $response && $this->applicationsExistFor( $this->wg->CityId, $response ) ) {
+			$mobileApplications = json_decode( $response, true );
+			$mobileApplicationsLinks = $this->traverseThrough( $mobileApplications );
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	/**
+	 * @return string or boolean (because of Http::request)
+	 */
+	private function fetchMobileApplicationsDetails() {
+		// currently it does not matter if Android or iOS value is added, data is returned for both Android and iOS
+		return Http::request( "GET",  'https://services.wikia.com/mobile-applications/platform/android' );
+	}
+
+
+	/**
+	 * @param string siteId - current site id
+	 * @param string response - raw, not parsed response
+	 *
+	 * @return boolean
+	 */
+	private function applicationsExistFor( $siteId, $response ) {
+		return strpos( $response, 'wikia_id":' . $siteId ) !== false;
+	}
+
+	/**
+	 * @param array $mobileApplications - array containing information about mobile applications
+	 * @return array - containing mobile applications links
+	 */
+	private function traverseThrough( $mobileApplications ) {
+		$mobileApplicationsLinks = [];
+		$siteId = $this->wg->CityId;
+
+		foreach ( $mobileApplications['apps'] as $app ) {
+			foreach ( $app['languages'] as $language ) {
+				if ( $language['wikia_id'] ==  $siteId) {
+					if ( $app['android_release'] ) {
+						$release = $app['android_release'];
+						$mobileApplicationsLinks['android'] = "https://play.google.com/store/apps/details?id=$release";
+					}
+					if ( $app['ios_release'] ) {
+						$release = $app['ios_release'];
+						$mobileApplicationsLinks['ios'] = "https://itunes.apple.com/us/app/id$release";
+					}
+					break;
+				}
+			}
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
 	/**
 	 * TODO Move this into unsubscribe extension?
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this "TODO" comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)